addons/repo-proxy/ftests/src/main/java/org/commonjava/indy/repo/proxy/ftest/RepoProxyMethodsTest.java

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ */
+package org.commonjava.indy.repo.proxy.ftest;
+
+import org.apache.commons.io.IOUtils;
+import org.commonjava.indy.client.core.IndyClientException;
+import org.commonjava.indy.ftest.core.AbstractContentManagementTest;
+import org.commonjava.indy.model.core.HostedRepository;
+import org.commonjava.indy.model.core.RemoteRepository;
+import org.commonjava.indy.pkg.maven.model.MavenPackageTypeDescriptor;
+import org.commonjava.indy.test.fixture.core.CoreServerFixture;
+import org.hamcrest.CoreMatchers;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+/**
+ * Check if the repo proxy addon can proxy a hosted repo to a remote repo with specific api methods
+ * <br/>
+ * GIVEN:
+ * <ul>
+ *     <li>Hosted and Remote Repo</li>
+ *     <li>Hosted does not contain pathA but remote does</li>
+ *     <li>Configured the repo-proxy disabled</li>
+ *     <li>Configured proxy rule with mapping hosted proxy to remote</li>
+ *     <li>Configured api methods</li>
+ * </ul>
+ * <br/>
+ * WHEN:
+ * <ul>
+ *     <li>Request pathA through hosted with specified api methods</li>
+ *     <li>Request pathA through hosted without specified api methods</li>
+ * </ul>
+ * <br/>
+ * THEN:
+ * <ul>
+ *     <li>The content of pathA can be returned correctly from hosted when methods applied</li>
+ *     <li>The content of pathA can be returned correctly from hosted when methods not applied</li>
+ * </ul>
+ */
+public class RepoProxyMethodsTest
+        extends AbstractContentManagementTest
+{
+    private static final String REPO_NAME = "pnc-builds";
+
+    private HostedRepository hosted;
+
+    private RemoteRepository remote;
+
+    private static final String PATH1 = "foo/bar/1.0/foo-bar-1.0.txt";
+
+    private static final String CONTENT1 = "This is content 1";
+
+    @Before
+    public void setupRepos()
+            throws Exception
+    {
+
+        server.expect( server.formatUrl( REPO_NAME, PATH1 ), 200, new ByteArrayInputStream( CONTENT1.getBytes() ) );
+
+        remote = client.stores()
+                       .create( new RemoteRepository( MavenPackageTypeDescriptor.MAVEN_PKG_KEY, REPO_NAME,
+                                                      server.formatUrl( REPO_NAME ) ), "remote pnc-builds",
+                                RemoteRepository.class );
+
+        hosted = client.stores()
+                       .create( new HostedRepository( MavenPackageTypeDescriptor.MAVEN_PKG_KEY, REPO_NAME ),
+                                "hosted pnc-builds", HostedRepository.class );
+    }
+
+    @Test
+    public void run()
```

Comment:
Another approach here would be to use two test methods, and in one use @Expect(IndyClientException)

It might give faster feedback in the test result too, I guess. But it's not a big deal either way.

Reply:
Will do.